### DISCUSSION TITLE
挂载arex java agent后，服务端启动慢了很多

### DISCUSSION THREAD

**Discussion by liuliuwd** [arextest/arex-agent-java#543] [General]
挂载arex java agent前，服务端启动耗时15s左右，挂载后启动耗时1分10s左右

**Comment by lucas-myx** [comment]
Hello, sorry for the late reply!   

When the arex-java-agent starts, it will detect and modify the framework source code of about a more than ten modules,  
such as `Spring, Redis, MyBatis, Dubbo, HttpClient, Mongo, Netty, JDK excutors`,   etc., add record replay stub code,   
will trigger JVM **retansform** and **recompile** class, and this process depends on how many open source frameworks are used in the project.  

This process is time-consuming, and we will add metric functionality in the future to make users more aware of the specific time for modifying each module.  

At the same time, you can also check the documentation: [FAQ](https://doc.arextest.com/docs/FAQ)
You can also directly raise [issue](https://github.com/arextest/arex-agent-java/issues) for us.
